gc/gc-base/src/main/java/org/projectnessie/gc/base/DistributedIdentifyContentsExecutor.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.base;
+
+import java.io.Serializable;
+import java.lang.reflect.InvocationTargetException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import javax.validation.constraints.NotNull;
+import org.apache.commons.lang3.mutable.MutableBoolean;
+import org.apache.spark.sql.SparkSession;
+import org.projectnessie.client.NessieClientBuilder;
+import org.projectnessie.client.StreamingUtil;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.client.http.HttpClientBuilder;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.Detached;
+import org.projectnessie.model.LogResponse;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains the methods that executes in spark executor for {@link
+ * GCImpl#identifyExpiredContents(SparkSession)}.
+ */
+public class DistributedIdentifyContentsExecutor implements Serializable {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(DistributedIdentifyContentsExecutor.class);
+  private final Map<String, String> config;
+
+  public DistributedIdentifyContentsExecutor(Map<String, String> config) {
+    this.config = config;
+  }
+
+  /**
+   * Builds the client builder; default({@link HttpClientBuilder}) or custom, based on the
+   * configuration provided.
+   *
+   * @param configuration map of client builder configurations.
+   * @return {@link NessieApiV1} object.
+   */
+  protected static NessieApiV1 getApi(Map<String, String> configuration) {
```

Comment:
should be moved to GCUtil as it is used from this class and DistributedIdentifyContents